### PR TITLE
feat: skip cells which has data or type when calculate balance

### PR DESCRIPTION
**Pull request by classicalliu**
Now the `balance` not include cells which has data or type, and `totalBalance` includes.
Has a setting to set can use cells with data/type or not.

### CODE REVIEW

File: packages/neuron-wallet/src/services/skip-data-and-type.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+  static getInstance(): SkipDataAndType {
+    if (!SkipDataAndType.instance) {
+      SkipDataAndType.instance = new SkipDataAndType()
+    }
+
+    return SkipDataAndType.instance
+  }
+
+  // open means can use cells with data and type
```
Comment (Keith-CY) [inline packages/neuron-wallet/src/services/skip-data-and-type.ts]: Where is the `open` used?

File: packages/neuron-wallet/src/services/skip-data-and-type.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      JSON.stringify({
+        skip,
+      })
+    )
+    // cache this variable
+    this.skip = skip
+  }
+
+  public get(): boolean {
+    // if cached, don't to read file
```
Comment (Keith-CY) [inline packages/neuron-wallet/src/services/skip-data-and-type.ts]:
```suggestion
    // if cached, don't read file
```

File: packages/neuron-wallet/src/services/skip-data-and-type.ts (hunk 1 above)
Comment (classicalliu) [reply packages/neuron-wallet/src/services/skip-data-and-type.ts]: Renamed to `skip` ...

File: packages/neuron-wallet/src/database/address/dao.ts [hunk 3]
```diff
@@ -67,9 +70,12 @@ export default class AddressDao {
         const addressEntity = entity
         addressEntity.txCount = txCount
         const lockHashes: string[] = await LockUtils.addressToAllLockHashes(addressEntity.address)
-        addressEntity.liveBalance = await CellsService.getBalance(lockHashes, OutputStatus.Live)
-        addressEntity.sentBalance = await CellsService.getBalance(lockHashes, OutputStatus.Sent)
-        addressEntity.pendingBalance = await CellsService.getBalance(lockHashes, OutputStatus.Pending)
+        addressEntity.liveBalance = await CellsService.getBalance(lockHashes, OutputStatus.Live, true)
```
[the remark this answers is not in the document]
Comment (classicalliu) [reply packages/neuron-wallet/src/database/address/dao.ts]: No need, it's not slow and avoid database timeout ...

File: packages/neuron-wallet/src/services/skip-data-and-type.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
+import FileService from './file'
+
+export default class SkipDataAndType {
+  private static moduleName = ''
+  private static fileName = 'skip-data-and-type.json'
```
Comment (ashchan) [inline packages/neuron-wallet/src/services/skip-data-and-type.ts]: How about a global `settings.json` for all related persisted app configuration? I don't see any benefit maintaining a single file for each feature.

Comment (classicalliu) [reply packages/neuron-wallet/src/services/skip-data-and-type.ts]: Add a `BaseSettings` to read/update settings and read/update `SkipDataAndType` via this.

**Pull request merged**